src/pivot-table/hooks/use-column-width.ts

```diff
@@ -1,41 +1,65 @@
 import { memoize } from "qlik-chart-modules";
 import { useCallback, useMemo } from "react";
 import { PSEUDO_DIMENSION_INDEX } from "../../constants";
-import type { LayoutService, LeftDimensionData, Rect } from "../../types/types";
+import { ColumnWidthType, type ExtendedNxDimensionInfo, type ExtendedNxMeasureInfo } from "../../types/QIX";
+import type {
+  DimensionContentStyling,
+  HeaderStyling,
+  LayoutService,
+  LeftDimensionData,
+  MeasureContentStyling,
+  Rect,
+  TopDimensionData,
+} from "../../types/types";
 import { GRID_BORDER } from "../constants";
 import { useStyleContext } from "../contexts/StyleProvider";
 import useMeasureText from "./use-measure-text";
 
 interface ColumnWidthHook {
   leftGridWidth: number;
   rightGridWidth: number;
-  totalMeasureInfoColumnWidth: number;
   getLeftColumnWidth: (index: number) => number;
-  getDataColumnWidth: (index: number) => number;
   getMeasureInfoWidth: (index: number) => number;
-  getTotalWidth: () => number;
+  getLeafWidth: (index?: number) => number;
+  totalWidth: number;
+  showRightBorder: boolean;
 }
 
 export const EXPAND_ICON_WIDTH = 30;
-const MIN_COLUMN_WIDTH = 100;
+const MIN_COLUMN_WIDTH = 30;
 const MAX_RATIO_OF_TOTAL_WIDTH = 0.75;
+const DEFAULT_PIXEL_VALUE = 200;
+const DEFAULT_PERCENTAGE_VALUE = 20;
+
+// TODO: refactor and move this to use-measure-text file
+const useMeasureAndEstimate = (styling: HeaderStyling | MeasureContentStyling | DimensionContentStyling) =>
```

Comment:
This is basically what you always do, so might as well just send in the object will the styling properties. Will look at a refactor, probably in a separate PR